napari/_qt/_qapp_model/_tests/test_processors.py

```diff
@@ -3,13 +3,46 @@
 
 import numpy as np
 import pytest
+from qtpy.QtWidgets import QWidget
 
 from napari._qt._qapp_model.injection._qprocessors import (
+    _add_future_data,
     _add_layer_data_to_viewer,
+    _add_layer_data_tuples_to_viewer,
+    _add_layer_to_viewer,
+    _add_plugin_dock_widget,
 )
 from napari.types import ImageData, LabelsData
 
 
+def test_add_plugin_dock_widget(qtbot):
+    widget = QWidget()
+    viewer = MagicMock()
```

Comment:
Are we able to use `make_napari_viewer` and then after we've added to the viewer, check that the layer has been added to the viewer?

Reply:
I think this test doesn't add a layer, maybe you mean like checking there is a dockwidget related with the created widget added to the main window here? Or maybe you mean like when adding a dockwidget it should be also tested that a layer can be added?

Reply:
Ignore this, sorry!

Reply:
Haha wait don't ignore it! I think it makes sense here to test that after processor runs, we have the widget added to the `viewer.window.qt_main_window._dock_widgets` list

Reply:
Sorry I was typing a longer comment: https://github.com/napari/napari/pull/7280#issuecomment-2434105302 :grimacing: